apis/v1alpha2/httproute_types.go

```diff
@@ -457,6 +461,13 @@ type HTTPRouteMatch struct {
 	//
 	// +optional
 	ExtensionRef *LocalObjectReference `json:"extensionRef,omitempty"`
+
+	// Method specifies HTTP method matchers. Multiple match
+	// values are ORed together, meaning, a request must match any of the
+	// specified methods to select the route.
+	//
+	// +optional
+	Method []HTTPMethod `json:"methods,omitempty"`
```

Comment:
I think I'd prefer starting with a single method here. If multiple methods are needed they can be communicated with multiple matches.